### PR TITLE
Defect/security annotations on abstracts

### PR DESCRIPTION
**Pull request by esotrope**
Fixes security rules on group components.
Group components incorrectly had PII_READ permission enforced
This will not affect dwtest@example.com because it had that permission anyway

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/group/DefaultGroupExamItemService.java [hunk 1]
```diff
@@ -25,7 +25,6 @@
         this.examItemRepository = examItemRepository;
     }
 
-    @Override
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/group/DefaultGroupExamItemService.java]: This isn't implementing the interface method?

Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/group/DefaultGroupExamItemService.java]: Do we wan't to have @Override for implementing methods? i found that it is inconsistently used so i arbitrarily removed it - can add it back

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/ExamItemService.java [hunk 2]
```diff
@@ -17,7 +17,6 @@
      * @param search the search parameters to limit the results by
      * @return all the exam item scores and item information for a given assessment
      */
-    @PreAuthorize("hasAuthority('PERM_PII_READ')")
     GroupExamItemScores getExamItemScores(@NotNull User user, @NotNull T search);
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/ExamItemService.java]: It's not obvious to me why exam item scores doesn't need protection?

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/group/DefaultGroupExamItemService.java (hunk 1 above)
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/group/DefaultGroupExamItemService.java]: Hmm. I always put @Override for implementing methods. I thought that was standard (?)

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/ExamItemService.java (hunk 2 above)
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/search/exam/ExamItemService.java]: Its a little hard to follow here but there are kind of "abstract" interfaces here and "concrete" interfaces - what i did here was move the security annotations from the abstract interfaces to the concrete in the packages because the security is package specific
